internal/controllers/gateway/backendtlspolicy_utils.go

```diff
@@ -175,3 +200,77 @@ func sortGateways(gateways []gatewayapi.Gateway, kicAncestors []gatewayapi.Polic
 		}
 	})
 }
+
+// validateBackendTLSPolicy validates the given BackendTLSPolicy and returns the accepted Condition related to the policy.
+func (r *BackendTLSPolicyReconciler) validateBackendTLSPolicy(ctx context.Context, policy gatewayapi.BackendTLSPolicy) (acceptedCondition *metav1.Condition, err error) {
+	acceptedCondition = &metav1.Condition{
+		Type:               string(gatewayapi.PolicyConditionAccepted),
+		Status:             metav1.ConditionTrue,
+		Reason:             string(gatewayapi.PolicyConditionAccepted),
+		ObservedGeneration: policy.Generation,
+	}
+
+	for _, targetRef := range policy.Spec.TargetRefs {
+		if (targetRef.Group != "core" && targetRef.Group != "") || targetRef.Kind != "Service" {
+			continue
+		}
+		policies := &gatewayapi.BackendTLSPolicyList{}
+		if err := r.List(ctx, policies,
+			client.InNamespace(policy.Namespace),
+			client.MatchingFields{backendTLSPolicyTargetRefIndexKey: string(targetRef.Name)},
+		); err != nil {
+			return nil, err
+		}
+
+		if len(policies.Items) > 1 {
+			acceptedCondition = &metav1.Condition{
+				Type:    string(gatewayapi.PolicyConditionAccepted),
+				Status:  metav1.ConditionFalse,
+				Reason:  string(gatewayapi.PolicyReasonConflicted),
+				Message: "Multiple BackendTLSPolicies target the same service",
+			}
+			return acceptedCondition, nil
+		}
+	}
+
+	var invalidMessage string
+	for _, caCert := range policy.Spec.Validation.CACertificateRefs {
+		if (caCert.Group != "core" && caCert.Group != "") || caCert.Kind != "ConfigMap" {
+			invalidMessage = "CACertificateRefs must reference ConfigMaps in the core group"
+			break
+		}
+	}
+	if len(policy.Spec.Validation.SubjectAltNames) > 0 {
+		if invalidMessage != "" {
+			invalidMessage += " - "
+		}
+		invalidMessage += "SubjectAltNames feature is not currently supported"
+	}
+	if policy.Spec.Validation.WellKnownCACertificates != nil {
+		if invalidMessage != "" {
+			invalidMessage += " - "
+		}
+		invalidMessage += "WellKnownCACertificates feature is not currently supported"
+	}
+	if invalidMessage != "" {
+		acceptedCondition.Status = metav1.ConditionFalse
+		acceptedCondition.Reason = string(gatewayapi.PolicyReasonInvalid)
+		acceptedCondition.Message = invalidMessage
+	}
+
+	return acceptedCondition, nil
+}
+
+// list namespaced names of configmaps referred by the gateway.
+func listConfigMapNamesReferredByBackendTLSPolicy(policy *gatewayapi.BackendTLSPolicy) map[k8stypes.NamespacedName]struct{} {
+	// no need to check group and kind, as if they were different from core/Configmap, the policy would have been marked as invalid.
+	nsNames := make(map[k8stypes.NamespacedName]struct{})
```

Comment:
```suggestion
	nsNames := make(map[k8stypes.NamespacedName]struct{}, len(policy.Spec.Validation.CACertificateRefs))
```
You can prealocate here, it will have exactly this size